Library/Homebrew/tap.rb

```diff
@@ -146,9 +146,22 @@ def remote_repo
 
     return unless remote
 
-    @remote_repo ||= remote.delete_prefix("https://github.com/")
-                           .delete_prefix("git@github.com:")
-                           .delete_suffix(".git")
+    path_base = nil
+    begin
+      path_base = URI.parse(remote).path.delete_prefix("/")
```

Comment:
I think it's much less clear what's going on from here. I think the original code was much easier to follow for the `github.com` case so would be good to specifically use `HOMEBREW_GITHUB_API_URL`

Reply:
Even that's not terribly easy to follow from my perspective. It feels like this needs a dedicated environment variable with the internal GHES hostname which then can be checked here and call dedicated methods for different behaviour.